src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,452 @@
+ as the return type for asynchronous functions that don't return usable values.
+{{ site.alert.end }}
+
+Now that you have declared an async function:
+  1. Your function returns a future that completes with the type that you specify.
+  1. You must use the `await` keyword before invoking the function  to "wait"
+   for the future to complete either with a value or an error.
+
+{% prettify dart %}
+var order = [!await!] getUserOrder();
+{% endprettify %}
+
+
+The following example provides a side-by-side comparison of a synchronous function
+and an asynchronous function. Since you can only use the `await` keyword within 
+an `async` function body, the example converts the parent function from 
+synchronous to asynchronous thus enabling use of the `await` keyword: 
+
+
+[//]: https://gist.github.com/8f4d136e7eb447f4acc7ccb523006e10
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new.html?id=8f4d136e7eb447f4acc7ccb523006e10"
+  style="margin-top: 25px; margin-bottom: 25px"
+  frameborder="no"
+  height="425"
+  width="100%">
+</iframe>
+
+In this synchronous example the console immediately prints the messages 
+"Fetching user order..." and "Your order is: Large Latte".
+
+[//]: https://gist.github.com/b8d026d2b23102084534159fafc9d7c6
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new.html?id=b8d026d2b23102084534159fafc9d7c6"
+  style="margin-top: 25px; margin-bottom: 25px"
+  frameborder="no"
+  height="425"
+  width="100%">
+</iframe>
+
+In this asynchronous example, the console does not print "Your order is: Large
+Latte" until after a four second delay has passed. This asynchronous example has
+only four changes from the preceding synchronous example:
+
+1. Update the return types for `createOrderMessage` and `getUserOrder` from 
+`String` to `Future<String>`.
+2. Add the **`async`** keyword before the method bodies for `createOrderMessage`
+and the `main` method.
+3. Use the **`await`** keyword before calling the asynchronous functions 
+`getUserOrder` and `createOrderMessage`.
+4. Update the body of `getUserOrder` to use `Future.delayed` to mock asynchronous
+operations that fetch the user's order.
+
+
+{{ site.alert.info }}
+  Dart can [infer the Future\<T\> type](https://dart.dev/guides/language/sound-dart#type-inference) for you.
+{{ site.alert.end }}
+
+{{ site.alert.secondary }}
+  Quick review: 
+  * An `async` function returns a future.
+  * You can only use the `async` and `await` keywords within an `async` function body.
+  * To declare an `async` function, add the keyword after the function's parameters but before the function's body.
+  * Use the `await` keyword before invoking an async function to "wait" for the future to complete either with a value or an error.
+{{ site.alert.end }}
+
+## Execution flow with async and await
+
+As of Dart 2.0, functions marked as `async` run synchronously until the first
```

Comment:
We use “Dart 2”, not “Dart 2.0”.  https://docs.google.com/document/d/1vClxUPrWU1fsRq5Ws_KSz6cZHJ2QxkCCG436HJNNGh4/edit#heading=h.3ka2czxdh5qy